File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    
    const numer = (numer2 * denom1) + (numer1 * denom2);
    const denom = denom1 * denom2;
    
    //최대공약수를 구하는 함수
    //a와 b의 최대공약수는 b와 a를 b로 나눈 나머지의 최대공약수와 같다.
    function gcd(a, b) {
        return b === 0 ? a : gcd(b, a%b)
    }
    //구해진 최대공약수
    const common = gcd(numer, denom);
    
    return [numer / common, denom / common];
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        
        //분모 = denom, 분자 = numer
        int denom = denom1 * denom2;
        int numer = denom1 * numer2 + denom2 * numer1;
        
        //최대공약수 구하기
        int gcd = GCD(numer, denom);
        
        //분모, 분자에 최대공약수로 나누기, 새로운 배열 생성해서 바로 리턴
        return new int[]{numer/gcd, denom/gcd};
    }
    
    private static int GCD(int a, int b){
        while (b!=0){
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }
}
/*
알고리즘 설명
a, b 중에서 a가 더 큰 수라고 가정

a % b를 계산해서 나머지를 구함 → 이걸 tmp에 저장

a ← b, b ← tmp로 갱신

b가 0이 될 때까지 반복

반복 종료 시 a가 최대공약수
*/